apps/mobile/src/features/ConfirmTx/hooks/useTxSignerAutoSelection.ts

```diff
@@ -0,0 +1,22 @@
+import { useLayoutEffect } from 'react'
+import { MultisigExecutionDetails } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import { useTxSignerState } from '@/src/features/ConfirmTx/hooks/useTxSignerState'
+import { useTxSignerActions } from '@/src/features/ConfirmTx/hooks/useTxSignerActions'
+
+export const useTxSignerAutoSelection = (detailedExecutionInfo?: MultisigExecutionDetails) => {
+  const { activeTxSigner, appSigners, proposedSigner, hasSigned } = useTxSignerState(detailedExecutionInfo)
+  const { setTxSigner } = useTxSignerActions()
+
+  useLayoutEffect(() => {
+    if (proposedSigner && activeTxSigner?.value !== proposedSigner.value && hasSigned) {
```

Comment:
This condition is hard to understand. Is it possible to name it? What does proposedSigner mean? I would guess:

- activeTxSigner -> The currently selected signer
- hasSigned -> The currently selected signer has already signed
- proposedSigner -> ...?

Reply:
Yes, it's a brainfuck. Let's leave it as is now and refactor if it causes issues. When you add a signer, this signer is not auto applied as a signer for the txs. We do some "calculations" to figure out who is supposed to sign the tx. If you don't have an active signer, but have an imported signer and this signer is not the proposer of the tx and he is allowed to sign this tx, then this will be the proposedSidner and we will dispatch the event to set him as the activeSigner. I said, that it is a brainfuck, right?